src/box/index.h

```diff
@@ -458,6 +458,17 @@ int
 exact_key_validate(struct key_def *key_def, const char *key,
 		   uint32_t part_count);
 
+/**
+ * Check that index and key are valid for point lookup. Point lookup
+ * points at single tuple.
```

Comment:
Do we really need this new function? Can't we just patch `exact_key_validate` to take `index_def` and check that the index is actually unique? This would be consistent with `key_validate`.

I'd also drop `exact_key_validate_nullable` while we are at it (inline in `iterator_position_validate`).

Reply:
Done. Diff is bit messy.